main.js

```diff
@@ -1,100 +1,233 @@
+   */
+  display: function () {
+    for (const die of this.positions) {
+      this.positionValues[die] === 0
+        ? (dicePositions[die].src = this.images[0])
+        : (dicePositions[die].src = this.images[this.positionValues[die]]);
+    }
+  },
+
+  /**
+   * Dice selected will update shouldRoll to false to prevent rerolling that die.
+   */
+  updateShouldRoll: function () {
+    for (const die of this.positions) {
+      if (dicePositions[die].classList[1] === "select")
+        this.shouldRoll[die] = false;
+    }
+  },
+
+  /**
+   * Remove 'select' class from all dice
+   */
+  removeAllSelect: function () {
+    for (const die of this.positions) {
+      dicePositions[die].classList.remove("select");
+    }
+  },
 };
 
-// Display dice - if 0 show blank die else show die image of the value.
-const displayDice = () => {
-    for (const die in dice.positionValues) {
-    dice.positionValues[die] === 0
-    ? (dicePositions[die].src = `./images/blank_dice.png`)
-    : (dicePositions[die].src = `./images/${dice.positionValues[die]}_dice.png`);
-}
-};
+// --- SELECT --- \\
 
-// dice selected will change rollBoolean to false to prevent rerolling that die.
-const lockSelectedDiceOnRoll = () => {
-    for (let i = 0; i < dicePositions.length; i++) {
-        if (dicePositions[i].classList[1] === "select") dice.rollBoolean[i] = false
-    }
-    console.log(`boolean ${dice.rollBoolean}`)
-}
+// Event listers added on all dice for toggling the 'select' class
+for (const die of dice.positions) {
+  dicePositions[die].addEventListener("click", () => {
+    if (dice.shouldRoll[die] && dice.positionValues[0] !== 0) {
+      dicePositions[die].classList.toggle("select");
 
-// --- Select --- \\
-
-// Add event listener on each die.
-for (let i = 0; i < dicePositions.length; i++) {
-        dicePositions[i].addEventListener("click", () => {
-            // toggle Select as long as die is not blank or have "lock" as a class.
-            if (dice.positionValues[0] === 0) {
-                return
-            } else if (dice.rollBoolean[i]) {
-                dicePositions[i].classList.toggle("select");
-            }
-    });
+      points.selectedTotal = 0;
+      points.calculate();
+      uncommitedPoints.innerHTML = points.turnTotal + points.selectedTotal;
+    }
+  });
 }
 
-// --- Hold --- \\
+// --- HOLD --- \\
 
 // Hold Dice - Switch player
-document.getElementById("hold-button").addEventListener('click',() => {
-    const activePlayer_1 = document.getElementById("player1")
-    const activePlayer_2 = document.getElementById("player2")
-
-    //switch Player
-    activePlayer_1.classList.toggle("active")
-    activePlayer_2.classList.toggle("active")
-
-    //reset on roundTotal and selectedTotal points
-    points.roundTotal = 0
-    points.selectedTotal = 0
-
-})
+document.getElementById("hold-button").addEventListener("click", () => {
+  if (activePlayer1.classList[1] === "active") {
```

Comment:
This works for now. But it's a bit fragile. Maybe [`Array.includes`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) would be better?